src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -95,85 +94,51 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
             }
 
             var cloneReturnType = cloneMethod?.ReturnType;
-
-            var args = ArrayBuilder<(Symbol?, BoundExpression)>.GetInstance();
+            // PROTOTYPE: Handle dynamic
+            var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(
```

Comment:
>BoundObjectOrCollectionValuePlaceholder [](start = 39, length = 39)

Could we re-use `BindObjectInitializerExpression` for this portion of the method, or `BindObjectInitializerMemberAssignment`?

If not, I think we want to look at it and possibly bring over some additional logic:
- `            var objectInitializerMemberBinder = this.WithAdditionalFlags(BinderFlags.ObjectInitializerMember);`
- `                ReportDuplicateObjectMemberInitializers(boundMemberInitializer, memberNameMap, diagnostics);`

Reply:
No, object initializers are specifically more feature-full, and that is not intended to be supported.

Specifically, I think those flags are incorrect for this code.

Reply:
I think we're okay without `BinderFlags.ObjectInitializerMember`. It's just a diagnostic quality question.

You didn't address my question about calling `ReportDuplicateObjectMemberInitializers`. I think that `new Person() with { FIrst = null, First = null}` should produce a 1912 error (which we're currently missing).

I understand that object creation initializers are more feature-full, but that doesn't mean we shouldn't share code. 
It's easier to understand that we're doing the same thing as object creation (with some restrictions) rather than having a separate implementation, then it's not clear what the divergences may be.
We could bind and reject all the the limited case that we allow (no nesting, no indexing, only assignment expressions). Alternatively, we could pass in a new parameter "forWithExpression" into existing binding methods to restrict allowed cases.
Such code sharing would likely also help for handling `dynamic`, which existing object creation/initialization handles.

Reply:
That's true. Let me see what I can share.